web/src/user_status_ui.ts

```diff
@@ -1,19 +1,86 @@
 import $ from "jquery";
+import type * as tippy from "tippy.js";
 
 import render_set_status_overlay from "../templates/set_status_overlay.hbs";
 import render_status_emoji_selector from "../templates/status_emoji_selector.hbs";
 
 import * as dialog_widget from "./dialog_widget";
+import * as dropdown_widget from "./dropdown_widget";
 import * as emoji from "./emoji";
 import type {EmojiRenderingDetails} from "./emoji";
+import * as flatpickr from "./flatpickr";
 import {$t, $t_html} from "./i18n";
 import * as keydown_util from "./keydown_util";
 import * as people from "./people";
+import * as timerender from "./timerender";
 import * as user_status from "./user_status";
 import type {UserStatusEmojiInfo} from "./user_status";
 
 let selected_emoji_info: Partial<UserStatusEmojiInfo> = {};
 let default_status_messages_and_emoji_info: {status_text: string; emoji: EmojiRenderingDetails}[];
+let user_status_widget: dropdown_widget.DropdownWidget;
```

Comment:
maybe call it `clear_status_on_widget` because `user_status_widget` feels more like representing the whole UI to upadte user status. But this wiidget only cares about the dropdown thing.

Reply:
or, `status_expires_in_widget` as you are naming `StatusExpiresInWidgetOption` below to the options in the widget.

or better imo would be `scheduled_end_time_widget`/`option` -- just to use the term `scheduled_end_time` consistently instead of mixing `status_expiry_time` & `scheduled_end_time`.

Feel free to pick whatever feels good.

Reply:
ya, I guess `scheduled_end_time_widget` would be better, I'll update it.

Reply:
Updated.